src/Datagrid/SimplePager.php

```diff
@@ -66,36 +65,38 @@ public function getNbResults(): int
         return $n;
     }
 
-    public function getResults($hydrationMode = null)
+    public function getResults(?int $hydrationMode = null): array
```

Comment:
We still have a weird thing here, because the interface is: 
```
/**
     * Returns an array of results on the given page.
     *
     * @return array
     */
    public function getResults();
```

Reply:
Indeed , but I dont know if we can fix it now

Reply:
I created https://github.com/sonata-project/SonataAdminBundle/issues/6260 with the 4.0 milestone to not block your PR with this problem.

IMHO, if sonataAdmin works well with `getResults`, that means we could deprecate the argument in the storage bundles. Then it will possibily lead to a chain reaction of argument removal :)